tests/test_buffering.py

```diff
@@ -589,7 +589,7 @@ def test_conddim_w_shifting():
     op1 = Operator(eqns, opt='buffering')
 
     # Check generated code
-    assert len(retrieve_iteration_tree(op1)) == 3
+    assert len(retrieve_iteration_tree(op1)) == 4
```

Comment:
So what changed in the code since it seems to change existing tests not related to `init_to_host`

Reply:
There tests also generate initialisation code that is affected by the addition of the guard. The loop:

```
  if (time_m <= time_M)
  {
    /* Begin section0 */
    START_TIMER(section0)
    for (int db0 = 0; db0 <= 0; db0 += 1)
    {
      for (int x = x_m; x <= x_M; x += 1)
      {
        for (int y = y_m; y <= y_M; y += 1)
        {
          ub0[(db0 + (time_M / factor))%(1)][x][y] = u[db0 + (time_M / factor)][x][y];
        }
      }
    }
    STOP_TIMER(section0,timers)
    /* End section0 */
  }
```

becomes:

```
  if (time_m <= time_M)
  {
    /* Begin section0 */
    START_TIMER(section0)
    for (int db0 = 0; db0 <= 0; db0 += 1)
    {
      if (0 <= db0 + (time_M / factor))
      {
        for (int x = x_m; x <= x_M; x += 1)
        {
          for (int y = y_m; y <= y_M; y += 1)
          {
            ub0[(db0 + (time_M / factor))%(1)][x][y] = u[db0 + (time_M / factor)][x][y];
          }
        }
      }
    }
    STOP_TIMER(section0,timers)
    /* End section0 */
  }
```

Reply:
this is the iteration nest before adding the guard:

```
IterationTree(<WithProperties[affine,parallel]::Iteration db0; (0, 0, 1)>, <WithProperties[affine,parallel]::Iteration x; (x_m, x_M, 1)>, <WithProperties[affine,parallel]::Iteration y; (y_m, y_M, 1)>)
IterationTree(<WithProperties[affine,sequential]::Iteration time[t0,t1]; (time_m, time_M, 1)>,)
IterationTree(<WithProperties[affine,sequential]::Iteration time[t0,t1]; (time_m, time_M, 1)>, <WithProperties[affine,parallel,parallel=]::Iteration x; (x_m, x_M, 1)>, <WithProperties[affine,parallel,parallel=]::Iteration y; (y_m, y_M, 1)>)
```

and after adding the guard:

```
IterationTree(<WithProperties[affine,parallel]::Iteration db0; (0, 0, 1)>,)
IterationTree(<WithProperties[affine,parallel]::Iteration db0; (0, 0, 1)>, <WithProperties[affine,parallel]::Iteration x; (x_m, x_M, 1)>, <WithProperties[affine,parallel]::Iteration y; (y_m, y_M, 1)>)
IterationTree(<WithProperties[affine,sequential]::Iteration time[t0,t1]; (time_m, time_M, 1)>,)
IterationTree(<WithProperties[affine,sequential]::Iteration time[t0,t1]; (time_m, time_M, 1)>, <WithProperties[affine,parallel,parallel=]::Iteration x; (x_m, x_M, 1)>, <WithProperties[affine,parallel,parallel=]::Iteration y; (y_m, y_M, 1)>)
```

Iteration tree `IterationTree(<WithProperties[affine,parallel]::Iteration db0; (0, 0, 1)>,)` appears as a result. Not sure if this is expected or if you want me to probe into it.

Reply:
no, it's OK!